core/os/memory.h

```diff
@@ -122,10 +122,10 @@ _ALWAYS_INLINE_ T *_post_initialize(T *p_obj) {
 	return p_obj;
 }
 
-#define memnew(m_class) _post_initialize(new ("") m_class)
+#define memnew(m_class) _post_initialize(::new ("") m_class)
```

Comment:
Seems fine, but I'm curious on what was the issue there caused specifically by adding Jolt?

Reply:
Thanks for the detailed explanation!

It seems fine to me to change `memory.h` to add `::` for disambiguation.

Whether Jolt would benefit from also defining placement new operators to avoid such issues in the future, I'll leave to Jorrit to decide, but this shouldn't block this PR either way.

Reply:
> However, I just now found that you can actually define a placement new on a per-class basis as well, which I had no idea about, by simply doing this:
> 
> ```c++
> void *operator new (size_t size, void *ptr) noexcept { return ptr; }
> ```
> 
> ... which I must admit looks a bit strange, but adding that to Jolt's `JPH_OVERRIDE_NEW_DELETE` macro does seem to resolve the compilation error as well, without needing to touch `memory.h`.

I wasn't aware that this was possible, it is a much nicer solution than having to use `::new` everywhere. I've created a PR that implements this jrouwe/JoltPhysics#1385.

> (There's probably some discussion to be had regarding Godot's containers not respecting the alignment of its contained items as well, which I imagine makes them ill-suited for storing a lot of Jolt's types in the first place, since a lot of them are 16-byte aligned or more due to its use of SIMD.)

I'm more afraid of `std::atomic` and what a non-aligned member would do in that case. I'm hoping that the godot heap allocator only hands out 16 byte aligned addresses like the normal Windows/Linux allocator. If this is the case then there shouldn't really be an issue. Only in double precision mode you would align `RVec3`'s to 16 bytes instead of 32 bytes which may incur a small (probably unmeasurable) performance overhead.